pkgs/development/python-modules/gpt4all/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, stdenv
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+, requests
+, tqdm
+}:
+
+buildPythonPackage rec {
+  pname = "gpt4all";
+  version = "1.0.8";
+  format = "wheel";
+
+  disabled = pythonOlder "3.8";
+
+  src = let
+    system = {
+      "x86_64-linux" = {
+        name = "x86_64";
+        hash = "sha256-lrX44Tl4TZzhGuosttOdeGaUPXy1dJF69I7KTMyzGhI=";
+      };
+    }.${stdenv.system} or (throw "Unsupported system");
+  in fetchPypi {
+    inherit pname format version;
+    inherit (system) hash;
+    platform = "manylinux1_${system.name}";
+    dist = "py3";
+    python = "py3";
+  };
+
+  nativeBuildInputs = [
+    requests
+    tqdm
+  ];
+
```

Comment:
Can we run https://github.com/nomic-ai/gpt4all/tree/main/gpt4all-bindings/python/gpt4all/tests ?

Reply:
It seems the tests need to download things. I have no idea if those files could be included in the package. Maybe they are huge and maybe gpt4all can keep them up-to-date or something.

```
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED gpt4all/tests/test_gpt4all.py::test_inference - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
FAILED gpt4all/tests/test_gpt4all.py::test_inference_long_orca_3b - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
FAILED gpt4all/tests/test_gpt4all.py::test_inference_long_falcon - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
FAILED gpt4all/tests/test_gpt4all.py::test_inference_long_llama_7b - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
FAILED gpt4all/tests/test_gpt4all.py::test_inference_long_llama_13b - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
FAILED gpt4all/tests/test_gpt4all.py::test_inference_long_mpt - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
FAILED gpt4all/tests/test_gpt4all.py::test_inference_long_replit - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
FAILED gpt4all/tests/test_gpt4all.py::test_inference_long_groovy - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
FAILED gpt4all/tests/test_gpt4all.py::test_inference_hparams - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
FAILED gpt4all/tests/test_gpt4all.py::test_inference_falcon - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
FAILED gpt4all/tests/test_gpt4all.py::test_inference_mpt - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
FAILED gpt4all/tests/test_gpt4all.py::test_embedding - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
FAILED gpt4all/tests/test_gpt4all.py::test_empty_embedding - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
FAILED gpt4all/tests/test_gpt4all.py::test_download_model - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
======================== 14 failed, 1 warning in 2.58s =========================
/nix/store/3pfjacvdg9f491lfqc9qb2d0nknx73fb-stdenv-linux/setup: line 1596: pop_var_context: head of shell_variables not a function context
error: builder for '/nix/store/4bdn35chzhx4gyhssh7lhpaazzqr440g-python3.10-gpt4all-1.0.8.drv' failed with exit code 1;
       last 10 log lines:
       > FAILED gpt4all/tests/test_gpt4all.py::test_inference_long_replit - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
       > FAILED gpt4all/tests/test_gpt4all.py::test_inference_long_groovy - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
       > FAILED gpt4all/tests/test_gpt4all.py::test_inference_hparams - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
       > FAILED gpt4all/tests/test_gpt4all.py::test_inference_falcon - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
       > FAILED gpt4all/tests/test_gpt4all.py::test_inference_mpt - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
       > FAILED gpt4all/tests/test_gpt4all.py::test_embedding - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
       > FAILED gpt4all/tests/test_gpt4all.py::test_empty_embedding - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
       > FAILED gpt4all/tests/test_gpt4all.py::test_download_model - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='gpt4all.io',...
       > ======================== 14 failed, 1 warning in 2.58s =========================
       > /nix/store/3pfjacvdg9f491lfqc9qb2d0nknx73fb-stdenv-linux/setup: line 1596: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/4bdn35chzhx4gyhssh7lhpaazzqr440g-python3.10-gpt4all-1.0.8.drv'.
```

Reply:
Tests that download things should be deactivated.

Reply:
All 14 tests do :(

Reply:
> All 14 tests do :(

Is it possible to predownload them?

Reply:
> > All 14 tests do :(
> 
> Is it possible to predownload them?

If not, a comment nead `doCheck` would be nice.